src/drivers/mediatek/mt8195/interrupt.c

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Mediatek
+//
+// Author: Allen-KH Cheng <allen-kh.cheng@mediatek.com>
+
+#include <sof/common.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/io.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/spinlock.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/* fa00558c-d653-4851-a03a-b21f125a9524 */
+DECLARE_SOF_UUID("irq-mt8195", irq_mt8195_uuid, 0xfa00558c, 0xd653, 0x4851,
+		 0xa0, 0x3a, 0xb2, 0x1f, 0x12, 0x5a, 0x95, 0x24);
+
+DECLARE_TR_CTX(int_tr, SOF_UUID(irq_mt8195_uuid), LOG_LEVEL_INFO);
+
+/* os timer reg value * 77ns 13M os timer
+ * 1  ms: 12987* 1.5ms: 19436
+ * 2  ms: 25974* 3  ms: 38961
+ */
+
+static void irq_mask_all(void)
+{
+	io_reg_update_bits(RG_DSP_IRQ_EN, 0xffffffff, 0x0);
+	io_reg_update_bits(DSP_IRQ_EN, 0xffffffff, 0x0);
+}
+
+static void mtk_irq_mask(struct irq_desc *desc, uint32_t irq, unsigned int core)
+{
+	int32_t mask, in_irq, level;
+
+	if (!desc) {
+		level = GET_INTLEVEL(irq);
+		in_irq = GET_INTERRUPT_ID(irq);
+		if (level > MAX_IRQ_NUM)
+			tr_err(&int_tr, "Invalid interrupt %d", irq);
```

Comment:
If you have an invalid interrupt, you should return here.
Is there any reason why not?